ament_index_cpp/CMakeLists.txt

```diff
@@ -14,9 +14,12 @@ add_library(${PROJECT_NAME} SHARED
   src/get_search_paths.cpp
   src/has_resource.cpp)
 target_compile_definitions(${PROJECT_NAME} PRIVATE "AMENT_INDEX_CPP_BUILDING_DLL")
-target_include_directories(${PROJECT_NAME} PUBLIC include)
+target_include_directories(${PROJECT_NAME} PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include>")
 
 ament_export_include_directories(include)
+ament_export_interfaces(export_${PROJECT_NAME})
```

Comment:
Even after reading the documentation for `ament_export_interfaces()` I can't tell, is this an input variable or an output variable? Is `export_${PROJECT_NAME})` set before this is called?